core/sdk-core/src/main/java/software/amazon/awssdk/core/SplitTransformerConfiguration.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core;
+
+import java.util.Objects;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.core.async.AsyncResponseTransformer;
+import software.amazon.awssdk.core.internal.async.SplittingTransformer;
+import software.amazon.awssdk.utils.Validate;
+import software.amazon.awssdk.utils.builder.CopyableBuilder;
+import software.amazon.awssdk.utils.builder.ToCopyableBuilder;
+
+/**
+ * Configuration options for {@link AsyncResponseTransformer#split(SplitTransformerConfiguration)} to configure how the SDK
+ * should split the {@link AsyncResponseTransformer}.
+ *
```

Comment:
This description makes me thing the name is not quite right. "to configure __how__ the SDK should split..." sounds like it's configuration for the splitting transformer (the one we're calling `split` on), but the name makes it sound like it's for the `SplitAsyncResponseTransformer`, which is the result of the `split`.

@zoewangg WDYT?

Reply:
right, its used to instantiate a `SplittingTransformer` which is then passed to the `SplitAsyncResponseTransformer`

`SplittingTransformerConfiguration`?

Reply:
Okay thanks for clarifying. In that case then yeah I vote for `SplittingTransformerConfiguration`